mw_organisation_activity.go

```diff
@@ -9,8 +9,12 @@ import (
 	"github.com/TykTechnologies/tyk/config"
 )
 
-var orgChanMap = make(map[string]chan bool)
+type orgChanMapMu struct {
+	sync.RWMutex
+	channels map[string](chan bool)
+}
 
+var orgChanMap = orgChanMapMu{channels: map[string](chan bool){}}
```

Comment:
As a note on mutex vs `sync.Map`, although in this case the map is append-only and hence kind of works of the latter, [some benchmarks show](https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c) that it only performs better during very high read contention over 4-8+ real cores. In other cases it significantly looses to the mutex-based maps. At the same time `sync.Map` looses type-safety so I decided against this premature optimisation and I'm not sure it fits `orgActiveMap` either (it's not at all append-only as the sentinel goroutine writes at the same keys all the time).

Reply:
we had a conversation about `sync.Map` a while ago, it was added for good reason but breaks backward comparability with older versions of Go.

according to change: if we wrap mutex and protected map into struct - I think it is a good chance to add to this struct `Get` and `Set` methods (with all lock/defers inside them) to simplify code and prevent (at least logically) from using mutexes/maps directly

Reply:
Thanks for the suggestion but I think this needs explicit locking control (see another comment on the PR).